### PR TITLE
Process Shortcodes in WooCommerce custom templates - MAILPOET-5110

### PR DESCRIPTION
**Pull request by triple0t**
## Description

 Process Shortcodes in WooCommerce custom templates 

## Code review notes

_N/A_

## QA notes

You can test with the Date or Site shortcodes.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5110](https://mailpoet.atlassian.net/browse/MAILPOET-5110)

## After-merge notes

_N/A_

[MAILPOET-5110]: https://mailpoet.atlassian.net/browse/MAILPOET-5110?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 namespace MailPoet\WooCommerce\TransactionalEmails;
 
+use MailPoet\Cron\Workers\SendingQueue\Tasks\Shortcodes;
```
Comment (alex-mailpoet) [inline mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php]: It's a bit confusing that a cron-related class is used here. Can we use `MailPoet\Newsletter\Shortcodes\Shortcodes` instead?

File: mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php [hunk 2]
```diff
@@ -33,8 +34,11 @@ public function __construct(
   }
 
   public function render(NewsletterEntity $newsletter, ?string $subject = null) {
-    $renderedHtml = $this->renderer->renderAsPreview($newsletter, 'html', $subject);
+    $renderedNewsletter = $this->renderer->renderAsPreview($newsletter, 'html', $subject);
     $headingText = $subject ?? '';
+
+    $renderedHtml = Shortcodes::process($renderedNewsletter, null, $newsletter, null, null);
```
Comment (alex-mailpoet) [inline mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php]: This doesn't follow dependency injection. Can we inject the `MailPoet\Newsletter\Shortcodes\Shortcodes` class in the constructor and use it here afterwards?

**Review by alex-mailpoet — COMMENTED** [review]
See my remarks. And I have a question: not all shortcodes will work in WooCommerce core emails, e.g. subscriber & newsletter categories won't, but all of them are displayed in the editor modal. Should we hide them from WC emails, or is it not a big deal?

File: mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php (hunk 1 above)
Comment (triple0t) [reply mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php]: Done

File: mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php (hunk 2 above)
Comment (triple0t) [reply mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php]: Done

**Comment by triple0t** [comment]
Thank you for the review, @alex-mailpoet  

> See my remarks. And I have a question: not all shortcodes will work in WooCommerce core emails, e.g. subscriber & newsletter categories won't, but all of them are displayed in the editor modal. Should we hide them from WC emails, or is it not a big deal?

1. I don't think it's a big deal. 
2. Curious: Are we currently doing this somewhere in the codebase?

File: mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php [hunk 3]
```diff
@@ -77,4 +82,11 @@ public function prefixCss($css) {
     $print = $this->cssParser->print;
     return $print->plain();
   }
+
+  public function processShortcodes(NewsletterEntity $newsletter, $content) {
```
Comment (alex-mailpoet) [inline mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php]: I'd mark this function private, it isn't used anywhere else, neither in tests.

**Comment by alex-mailpoet** [comment]
> Curious: Are we currently doing this somewhere in the codebase?

I don't think we do. Okay, let's leave it like that for now.

File: mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php (hunk 3 above)
Comment (triple0t) [reply mailpoet/lib/WooCommerce/TransactionalEmails/Renderer.php]: Done

**Review by alex-mailpoet — APPROVED** [review]
Good job! Thanks!

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**